playbooks/service-provision.yaml

```diff
@@ -116,54 +241,74 @@
-        "{{ resource_claim_name }}" in namespace "{{ user_namespace }}"
-
-  - name: Wait for the Tower deployerJob to start
-    k8s_info:
-      kubeconfig: "{{ kubeconfig }}"
-      api_version: poolboy.gpte.redhat.com/v1
-      kind: ResourceClaim
-      namespace: "{{ user_namespace }}"
-      name: "{{ resource_claim_name }}"
-    register: r_claim
-    retries: "{{ wait_retries | default(30) }}"
-    delay: 2
-    until: >-
-      r_claim.resources|length == 1
-      and 'status' in r_claim.resources[0]
-      and 'resources' in r_claim.resources[0].status
-      and r_claim.resources[0].status.resources|length > 0
-      and 'state' in r_claim.resources[0].status.resources[0]
-      and 'status' in r_claim.resources[0].status.resources[0].state
-      and 'towerJobs' in r_claim.resources[0].status.resources[0].state.status
-      and 'provision' in r_claim.resources[0].status.resources[0].state.status.towerJobs
-      and 'deployerJob' in (r_claim.resources[0].status.resources[0]
-                           .state.status.towerJobs.provision)
-
-  - name: Report babylon assigned GUID
-    debug:
-      msg: "babylon.guid: {{ r_claim.resources[0].status.resources[0].state.spec.vars.job_vars.guid }}"
+        babylon.currentState:
+        {{ r_claim.resources[0].status.resources[0].state.spec.vars.current_state }}
 
-  - name: Report babylon assigned UUID
-    debug:
-      msg: "babylon.uuid: {{ r_claim.resources[0].status.resources[0].state.spec.vars.job_vars.uuid }}"
+  - block:
+    - name: Wait for the Tower deployerJob to start
+      k8s_info:
+        kubeconfig: "{{ kubeconfig }}"
+        api_version: poolboy.gpte.redhat.com/v1
+        kind: ResourceClaim
+        namespace: "{{ user_namespace }}"
+        name: "{{ resource_claim_name }}"
+      register: r_claim
+      retries: "{{ wait_retries | default(30) }}"
+      delay: 2
+      until: >-
+        'deployerJob' in r_claim.resources[0].status.resources[0].state.status.towerJobs.provision | default({})
+    rescue:
+    - name: Report failure waiting for provisioning state
+      fail:
+        msg: |-
+          AnarchySubject {{ anarchy_subject_name }} has does not have a deployerJob in its status.
+          Check dark-tower job-runner:
+            https://{{ tower_hostname }}/#/jobs/playbook/{{ tower_launch_job }}
+          Check AnarchyRuns for errors processing callback event:
+            oc get anarchyruns -n anarchy-operator -l anarchy.gpte.redhat.com/subject={{ anarchy_subject_name }}
 
-  - name: Save Tower provision job
+  - name: Save Tower provision deployerJob
     set_fact:
       tower_job: >-
-        {{ r_claim.resources[0].status.resources[0]
-        .state.status.towerJobs.provision.deployerJob }}
+        {{ r_claim.resources[0].status.resources[0].state.status.towerJobs.provision.deployerJob }}
+
+  - name: Report deployerJob
+    debug:
+      msg: >-
+        babylon.deployerJob: {{ tower_job }}
```

Comment:
in https://github.com/redhat-gpe/OPEN_Admin/blob/master/OPENTLC-Deployer/OPEN-Deployer.agnosticv.sh#L720 we print those info as well.